pkg/provisioner.go

```diff
@@ -50,113 +51,94 @@ func (s *ProvisionerServer) ProvisionerCreateBucket(ctx context.Context,
+	bucketName := req.GetName()
 	klog.V(3).InfoS("Create Bucket", "name", bucketName)
 
-	err := s.S3Client.CreateBucket(s3.BucketName)
+	err := s.S3Client.CreateBucket(bucketName)
 	if err != nil {
 		// Check to see if the bucket already exists by above api
 		klog.ErrorS(err, "Bucket creation failed")
 		return nil, status.Error(codes.Internal, "Bucket creation failed")
 	}
-	klog.Infof("Successfully created Backend Bucket %s", s3.BucketName)
+	klog.Infof("Successfully created Backend Bucket %q", bucketName)
 
 	return &cosi.ProvisionerCreateBucketResponse{
-		BucketId: s3.BucketName,
+		BucketId: bucketName,
 	}, nil
 }
 
 func (s *ProvisionerServer) ProvisionerDeleteBucket(ctx context.Context,
 	req *cosi.ProvisionerDeleteBucketRequest) (*cosi.ProvisionerDeleteBucketResponse, error) {
-	/*	protocol := req.GetProtocol()
-		if protocol == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "Protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "Protocol is nil")
-		}
-		s3 := protocol.GetS3()
-		if s3 == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "S3 protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "S3 Protocol is nil")
-		}
+	klog.Infof("Delete bucket %q", req.GetBucketId())
+	if _, err := s.S3Client.DeleteBucket(req.GetBucketId()); err != nil {
+		klog.Info("failed to delete bucket %q", req.GetBucketId())
+		return nil, status.Error(codes.Internal, "Bucket deletion failed")
+	}
 
-		if _, err := s.S3Client.DeleteBucket(s3.BucketName); err != nil {
-			klog.Info("failed to delete bucket", s3.BucketName)
-			return nil, status.Error(codes.Internal, "Bucket deletion failed")
-		}
-	*/
 	return &cosi.ProvisionerDeleteBucketResponse{}, nil
 }
 
 func (s *ProvisionerServer) ProvisionerGrantBucketAccess(ctx context.Context,
 	req *cosi.ProvisionerGrantBucketAccessRequest) (*cosi.ProvisionerGrantBucketAccessResponse, error) {
-	/*	protocol := req.GetProtocol()
-		if protocol == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "Protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "Protocol is nil")
-		}
-		s3 := protocol.GetS3()
-		if s3 == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "S3 protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "S3 Protocol is nil")
-		}
-		user, err := s.radosgwAdminClient.CreateUser(radosgwapi.UserConfig{
-			UID:         req.GetAccountName(),
-			DisplayName: req.GetAccountName(),
-		})
-		if err != nil {
-			klog.Error("failed to create user", err)
-			return nil, status.Error(codes.Internal, "User creation failed")
-		}
-
-		policy, err := s.S3Client.GetBucketPolicy(s3.BucketName)
-		if err != nil {
-			if aerr, ok := err.(awserr.Error); ok {
-				if aerr.Code() != "NoSuchBucketPolicy" {
-					return nil, status.Error(codes.Internal, "fetching policy failed")
-				}
+	userName := req.GetAccountName()
+	bucketName := req.GetBucketId()
+	accessPolicy := req.GetAccessPolicy()
```

Comment:
I don't aee req.AccessPolicy being used anywhere below? Instead you get the s3 policy/

Reply:
@jeffvance yaa it is TODO please see L96